src/java/org/apache/cassandra/cql3/statements/schema/CreateIndexStatement.java

```diff
@@ -194,7 +210,10 @@ public Keyspaces apply(Keyspaces schema)
             throw ire("Index %s is a duplicate of existing index %s", index.name, equalIndex.name);
         }
 
-        TableMetadata newTable = table.withSwapped(table.indexes.with(index));
+        // All indexes on one table must use the same key_compression.
+        // The newly created index forces key_compression on the previous indexes.
```

Comment:
Do we want to emit a client warning about this?

Reply:
If I only knew how to do it ;)

Reply:
This way:
`ClientWarn.instance.warn("Dear user, please be aware that...")`

Reply:
Done